### PR TITLE
Fix exception thrown while loading maps from CARTO backend

### PR DESCRIPTION
**Pull request by jesusbotella**
This PR fixes an error produced by `SampleMapPanel` component while rendering because of some mismatching properties in our side while creating data to inject to Kepler.gl.

This is SampleMapPanel component, which is the component showing a preview of the dataset (label, description, data source...etc):
Screenshot 2019-11-06 at 16 12 01

I opted to show it even though we don't have much data to display, but we can remove it altogether if we want to.

Related to https://github.com/CartoDB/kepler.gl/issues/32.

### CODE REVIEW

File: examples/demo-app/src/utils/cloud-providers/carto.js [hunk 1]
```diff
@@ -124,11 +124,18 @@ function loadMap(queryParams) {
     dispatch(setLoadingMapStatus(true));
     carto.PublicStorageReader.getVisualization(username, mapId).then((result) => {
       // These are the options required for the action. For now, all datasets that come from CARTO are CSV
-      const options = result.datasets.map((dataset) => ({
+      const options = result.datasets.map((dataset) => {
+        const datasetId = dataset.name.split('keplergl_public_v0_')[1];
+
         // TODO: customStorage should return dataset name without prefix
```
Comment (neokore) [inline examples/demo-app/src/utils/cloud-providers/carto.js]: Better move this `split` below the `TODO` comment

File: examples/demo-app/src/utils/cloud-providers/carto.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ async function uploadFile({blob, name: fileName, isPublic = true}) {
   }, cartoDatasets, true);
 
   return ({
-    url: `demo/map/carto/${result.id}?owner=${cs.username}`
+    url: `demo/map/carto?mapId=${result.id}&owner=${cs.getSQLClient().username}`
```
Comment (neokore) [inline examples/demo-app/src/utils/cloud-providers/carto.js]: Here we can access `username` through `carto`.
```suggestion
    url: `demo/map/carto?mapId=${result.id}&owner=${carto.username}`
```

**Review by neokore — CHANGES_REQUESTED** [review]
Great! Just some small tweaks :wink:

**Comment by jesusbotella** [comment]
I'm gonna revert changes made within Kepler components given that we're going to avoid showing that modal

File: examples/demo-app/src/utils/cloud-providers/carto.js (hunk 1 above)
Comment (VictorVelarde) [reply examples/demo-app/src/utils/cloud-providers/carto.js]: I'm ok with this now, but this should be a public method on the custom storage, based on 'namespace' & 'version' parameters logic. That detail should be hidden away from kepler.gl

File: examples/demo-app/src/utils/cloud-providers/carto.js [hunk 3]
```diff
@@ -124,11 +124,19 @@ function loadMap(queryParams) {
     dispatch(setLoadingMapStatus(true));
     carto.PublicStorageReader.getVisualization(username, mapId).then((result) => {
       // These are the options required for the action. For now, all datasets that come from CARTO are CSV
-      const options = result.datasets.map((dataset) => ({
+      const options = result.datasets.map((dataset) => {
         // TODO: customStorage should return dataset name without prefix
```
Comment (VictorVelarde) [inline examples/demo-app/src/utils/cloud-providers/carto.js]: Use https://github.com/CartoDB/toolkit/pull/42 for the TODO?

**Review by neokore — APPROVED** [review]
Reviewed and tested... and looks good! :muscle:

**Review by VictorVelarde — CHANGES_REQUESTED** [review]
Using latest toolkit linked, persistence works fine (except the problem with 'CREATE or REPLACE function' (hopefully soon to be solved. 

BUT, @jesusbotella can you check the 'styles' are correctly persisted?. When loading again the URL, it looks like they are lost in my tests (also, it doesn't preserve zoom)

**Comment by neokore** [comment]
@VictorVelarde this caused by the "don't split names" change in toolkit, there is a PR with changes to solve it in https://github.com/CartoDB/kepler.gl/pull/34

**Comment by jesusbotella** [comment]
As @neokore said, we're saving them all in our storage, I've just checked it. If you installed latest toolkit version, the split name is conflicting with this.

I'm gonna merge both PRs to `feature/share-to-carto` branch and then we can test everything from that branch.

**Pull request merged**